src/esp/assets/managers/SceneAttributesManager.cpp

```diff
@@ -339,6 +300,45 @@ PhysicsSceneAttributes::ptr SceneAttributesManager::initNewAttribsInternal(
   return newAttributes;
 }  // SceneAttributesManager::initNewAttribsInternal
 
+void SceneAttributesManager::setDefaultFileNameBasedAttributes(
+    PhysicsSceneAttributes::ptr attributes,
+    bool setFrame,
+    const std::string& fileName,
+    std::function<void(int)> meshTypeSetter) {
+  // TODO : support future mesh-name specific type setting?
+  using Corrade::Utility::String::endsWith;
+
+  Magnum::Vector3 up, up1{0, 1, 0}, up2{0, 0, 1};
+  Magnum::Vector3 fwd, fwd1{0, 0, -1}, fwd2{0, 1, 0};
+
+  // set default origin and orientation values based on file name
+  // from AssetInfo::fromPath
+  up = up1;
+  fwd = fwd1;
+  if (endsWith(fileName, "_semantic.ply")) {
+    meshTypeSetter(static_cast<int>(AssetType::INSTANCE_MESH));
+  } else if (endsWith(fileName, "mesh.ply")) {
+    meshTypeSetter(static_cast<int>(AssetType::FRL_PTEX_MESH));
+    up = up2;
+    fwd = fwd2;
+  } else if (endsWith(fileName, "house.json")) {
+    meshTypeSetter(static_cast<int>(AssetType::SUNCG_SCENE));
+  } else if (endsWith(fileName, ".glb")) {
```

Comment:
Why are we associating glb files with Matterport3D here? Is this a backward-compatibility thing?

What's the recommended way for a user to specify her desired orientUp and orientFront when loading a glb?

Reply:
> Why are we associating glb files with Matterport3D here? Is this a backward-compatibility thing?

My understanding is that this is a legacy thing. We wrote support for MP3D by looking for glb files, for Replica by looking for ply files, for the dataset that shall not be named by looking for house.json files. 

CC: @msavva

Reply:
Correct, this is the backward compatibility for the AssetInfo::fromPath() function. The upcoming tutorial will demonstrate the preferred programmatic and file-based configuration options for scenes.

Reply:
That's right, this branching logic was hardcoded assuming particular datasets are only loaded from particular file types.  We really ought to clean this up and start to rely on correctly populating an `AssetInfo` to define asset type, coordinate frame conventions etc.

Reply:
Just a note, this currently is hardcoded only as a fallback if the user did not specify other values in their provided json config file.

Reply:
> What's the recommended way for a user to specify her desired orientUp and orientFront when loading a glb?

The scene importer allows for these fields to be set in the scene-descriptor json.  These only consume the defaults if the user has not set anything.